### PR TITLE
Aws services/bug/ma 1955

### PR DESCRIPTION
**Pull request by chandramishra**

### CODE REVIEW

File: aws-federation/env/dev_roleName [hunk 1]
```diff
@@ -0,0 +1 @@
+SungardAS-aws-services-feder-LambdaFunctionIAMRole-1GY1VIJ5NKQ5C
```
Comment (RajkumarShivage) [inline aws-federation/env/dev_roleName]: use a key-value pair. e.g. role='rolename'

Comment (bhargavapratik) [reply aws-federation/env/dev_roleName]: I cannot use it , as not able to parse json in Makefile

Comment (RajkumarShivage) [reply aws-federation/env/dev_roleName]: use this ROLE := $(shell awk -F'[/=]' '{print $$2}' env/${ENV}) to read value from env file

Comment (RajkumarShivage) [reply aws-federation/env/dev_roleName]: in env file use key-value pair like this
role=SungardAS-aws-services-feder-LambdaFunctionIAMRole-1GY1VIJ5NKQ5C

Comment (bhargavapratik) [reply aws-federation/env/dev_roleName]: ohk , yes I can use awk . doing now.

Comment (RajkumarShivage) [reply aws-federation/env/dev_roleName]: populate staging and prod files too

Comment (RajkumarShivage) [reply aws-federation/env/dev_roleName]: Makefile snippet

ifdef ENV
else

        ENV=dev;

endif

ifdef ROLE
else
        ROLE := $(shell awk -F'[/=]' '{print $$2}' env/${ENV})

endif

**Comment by RajkumarShivage** [comment]
Added Sonal, she has worked on Makefile before

File: aws-federation/env/dev_roleName (hunk 1 above)
Comment (RajkumarShivage) [reply aws-federation/env/dev_roleName]: Update readme

File: aws-federation/Makefile [hunk 2]
```diff
@@ -0,0 +1,39 @@
+
+ifdef AWS_REGION
+else
+	AWS_REGION := us-west-1
+endif
+
+ifdef ENV
+else
+
+	ENV='dev';
```
Comment (RajkumarShivage) [inline aws-federation/Makefile]: remove single quote

File: aws-federation/Makefile [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+endif
+
+ifdef ROLE
+else
+	data := $(shell awk -F'[/=]' '{print $$2}' env/${ENV})
```
Comment (RajkumarShivage) [inline aws-federation/Makefile]: assign value to ROLE

File: aws-federation/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+## How To Setup
+
+    $ make -e AWS_REGION=<region> ENV=<dev|qa|staging|prod> | ROLE=<roleName>
+    
+    For example:
+     $ make -e AWS_REGION=us-east-1 ENV=dev
+     
+     $ make -e AWS_REGION=us-east-1 ROLE="SungardAS-aws-services-feder-LambdaFunctionIAMRole-1TJ0PPEOJSUL8"
```
Comment (RajkumarShivage) [inline aws-federation/README.md]: do we need double quotes here? have you tested this?

File: aws-federation/env/prod [hunk 5]
```diff
@@ -0,0 +1 @@
+roleName=
```
Comment (RajkumarShivage) [inline aws-federation/env/prod]: value?

File: aws-federation/env/staging [hunk 6]
```diff
@@ -0,0 +1 @@
+roleName=
```
Comment (RajkumarShivage) [inline aws-federation/env/staging]: we have aws-federation lambda running in every environment in use-east region. use the execution role of it here.

File: aws-federation/env/prod (hunk 5 above)
Comment (bhargavapratik) [reply aws-federation/env/prod]: Need to check in Prod for value.

**Comment by chandramishra** [comment]
LGTM

File: aws-federation/build.f/run_update_code.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+var updator = new (require('../../lib/lambda_code_updator'))();
+
+var environment = process.env.NODE_ENV;
+var configFile = 'default';
+if (environment)  configFile = environment;
+
+var config = require('./particles-aws-federation/config/' + configFile);
+console.log(config.s3[0].aws.bucket);
+
+//var functionNames = ['awsconfigrules-security-group-checker','awsconfigrules-enable-rules','awsconfigrules-fetch-rules','awsconfigrules-sg-portrange-checker','awsconfigrules-delete-rules', 'awsconfigrules-ec2-create-trigger-db-update'];
```
Comment (SonalOjha) [inline aws-federation/build.f/run_update_code.js]: Remove commented lines

File: aws-federation/build.f/run_update_code.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+var functionNames = ['index'];
+
+var input = {
+  "region": null,
+  "bucketName": null,
+  "keyName": "particles/assets/aws-federation.zip",
+  "zipFile" : "aws-federation.zip",
+  "sourceFolder" : "../..",
+  //"src" : ["awsconfigrules/index_*.js", "awsconfigrules/json/*.json", "lib/flow_controller.js", "lib/aws/*.js", "lib/aws-promise/*.js"]
```
Comment (SonalOjha) [inline aws-federation/build.f/run_update_code.js]: Remove commented lines

File: aws-federation/build.f/run_update_code.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  "src" : ["index.js", "lib/flow_controller.js", "lib/aws/*.js", "lib/aws-promise/*.js"]
+}
+console.log(input);
+
+update(0, functionNames, input);
+
+function update(idx, functionNames, input) {
+  input.functionName = functionNames[idx];
+  updateFunction(0, config.s3, input, function(err, data) {
```
Comment (SonalOjha) [inline aws-federation/build.f/run_update_code.js]: What is this for?

File: aws-federation/build.f/run_update_code.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+      }
+    }
+    else {
+      console.log("Failed to update codes");
+      process.exit(1);
+    }
+  });
+}
+
+function updateFunction(idx, s3, input, cb) {
```
Comment (SonalOjha) [inline aws-federation/build.f/run_update_code.js]: Do we need this?

File: aws-federation/index.js [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+    if (err) {
+      console.log(err, err.stack);
+      callback(err, null);
+    }
+    else {
+      console.log("successfully assumed role, '" + role.roleArn + "'");
+      //console.log(data);
+      if (++idx == roles.length) {
+        console.log("successfully completed to assume all roles");
+        /*creds = new AWS.Credentials({
```
Comment (SonalOjha) [inline aws-federation/index.js]: Remove commented code

**Pull request merged**